### PR TITLE
Features/arbiter

### PR DESCRIPTION
**Pull request by danone771909**

### CODE REVIEW

File: battleship-logic/src/main/java/dreamteam/battleship/arbiter/Arbiter.java [hunk 1]
```diff
@@ -19,29 +18,14 @@
```
Comment (egolesor) [inline battleship-logic/src/main/java/dreamteam/battleship/arbiter/Arbiter.java]: Why are u initializing the list 2 time.

File: battleship-logic/src/main/java/dreamteam/battleship/arbiter/ArbiterImpl.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+     *
+     * @param shipList is passed in constructor. shipList will be created base on another class.
+     */
+    public ArbiterImpl(List<Ship> shipList) {
+        this.shipList = shipList;
+    }
+
+    public void removeShip(Ship ship) {
+        if(shipList.contains(ship))
+            shipList.remove(ship);
```
Comment (egolesor) [inline battleship-logic/src/main/java/dreamteam/battleship/arbiter/ArbiterImpl.java]: you can write 
`if(!shipList.remove(ship)){
throw new IllegalArgumentException("The ship isn't into shipList");
}`

File: battleship-logic/src/main/java/dreamteam/battleship/movement/MovementContainerImpl.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    public void addMovement(int field, boolean isDamaged) {
+        if(!containsMovement(field)) movements.put(field, isDamaged);
+    }
+
+    /**
+     *
+     * @param field - field on Board
+     * @return true if movements contains field and false when movements doesn't contain field
+     */
+    protected boolean containsMovement(int field) {
```
Comment (egolesor) [inline battleship-logic/src/main/java/dreamteam/battleship/movement/MovementContainerImpl.java]: I neeeeeeed this method make it public

File: battleship-logic/src/main/java/dreamteam/battleship/ship/ShipImpl.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
+package dreamteam.battleship.ship;
+
+/**
+ * Simple representation of our ships. Variables are protected to get access
+ * to test part in project. ShipImpl could have size between 1 and 4.
+ */
+public class ShipImpl implements Ship{
```
Comment (egolesor) [inline battleship-logic/src/main/java/dreamteam/battleship/ship/ShipImpl.java]: refactoring needed there are too many parameters in the ship.

File: battleship-logic/src/main/java/dreamteam/battleship/arbiter/Arbiter.java (hunk 1 above)
Comment (danone771909) [inline battleship-logic/src/main/java/dreamteam/battleship/arbiter/Arbiter.java]: Ok it's fixed

**Pull request merged**